frontend/src/container/NewWidget/RightContainer/Threshold/ColorSelector.tsx

```diff
@@ -0,0 +1,89 @@
+import './ColorSelector.styles.scss';
+
+import { DownOutlined } from '@ant-design/icons';
+import { Button, ColorPicker, Dropdown, Space } from 'antd';
+import { Color } from 'antd/es/color-picker';
+import { MenuProps } from 'antd/lib';
+import useDebounce from 'hooks/useDebounce';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+import CustomColor from './CustomColor';
+
+function ColorSelector({
+	thresholdColor = 'Red',
+	setColor,
+}: ColorSelectorProps): JSX.Element {
+	const [colorFromPicker, setColorFromPicker] = useState<string>('');
+
+	const debounceColor = useDebounce(colorFromPicker);
+
+	useEffect(() => {
+		if (debounceColor) {
+			setColor(debounceColor);
+		}
+	}, [debounceColor, setColor]);
+
+	const handleColorChange = (_: Color, hex: string): void => {
+		setColorFromPicker(hex);
+	};
+
+	const items: MenuProps['items'] = [
+		{
+			key: '1',
```

Comment:
keys can be set to the value of color / `colorPicker`

Reply:
This should be a hard-coded or read-only value, as the drop will get re-render as the key changes, I have update it with hard-coded color names.